File: backend/app/models/asset.py
from sqlalchemy import Column
from sqlalchemy import Integer
from sqlalchemy import String
from sqlalchemy import ForeignKey

from app.db.database import Base


class Asset(Base):

    __tablename__ = "assets"

    id = Column(
        Integer,
        primary_key=True,
        index=True
    )

    name = Column(
        String,
        nullable=False
    )

    asset_type = Column(
        String,
        nullable=False
    )

    owner = Column(
        String,
        nullable=False
    )

    criticality = Column(
        String,
        nullable=False
    )

    ip_address = Column(
        String,
        nullable=True
    )

    environment = Column(
        String,
        nullable=False
    )

    org_id = Column(
        Integer,
        ForeignKey("organizations.id"),
        nullable=False,
        index=True
    )

File: frontend/src/api/types/asset.ts
// Hand-maintained against backend/app/models/asset.py.
export interface Asset {
  id: number;
  name: string;
  asset_type: string;
  owner: string;
  criticality: string;
  ip_address: string | null;
  environment: string;
  org_id: number;
}

// GET /assets/{id}/vulnerabilities - a purpose-built dict from
// app/api/assets.py, not the full Vulnerability model.
export interface AssetVulnerabilitySummary {
  id: number;
  plugin_id: string | null;
  title: string;
  severity: string;
  cvss_score: number;
  status: string;
}

// GET /assets/{id}/summary
export interface AssetSummary {
  asset_id: number;
  asset_name: string;
  critical: number;
  high: number;
  medium: number;
  low: number;
  total: number;
}

export interface AssetListParams {
  limit?: number;
  offset?: number;
  sort_by?: 'id' | 'name' | 'criticality' | 'environment';
  order?: 'asc' | 'desc';
  criticality?: string;
  environment?: string;
  asset_type?: string;
}
